File: 01-pattern-problems/python/pattern2.py

# Author: Sourabh Ninawe
class Pattern2:
    def print_pattern2_method1(self, n):
        print("Pattern2:Method1")
        for i in range(n):
            for j in range(i+1):
                print('*', end='')
            print()

    #shorter implementation
    def print_pattern2_method2(self, n):
        print("Pattern2:Method2")
        for i in range(1, n+1):
            print('*' * i)

if __name__ == "__main__":
    while True:
        try:
            n = int(input("Enter the number of lines to print pattern 2 (integer, 0 or negative to exit): "))
        except ValueError:
            print("Please enter a valid integer.")
            continue
        if n <= 0:
            break
        else:
            p2 = Pattern2()
            p2.print_pattern2_method1(n)
            p2.print_pattern2_method2(n)

File: 01-pattern-problems/java/Pattern2.java
import java.util.Scanner;
/** 
 * @author Sourabh Ninawe
 */
class Pattern2{
    public void printPattern2(int n){
        for(int i=0;i<n;i++){
            for(int j=0;j<=i;j++){
                System.out.print("*");
            }
            System.out.println();
        }
    }
    public static void main(String[] args) {
        while(true){
            System.out.println("Enter the number of lines to print pattern 2 in integer and -1 to exit");
            Scanner sc = new Scanner(System.in);
            int n = sc.nextInt();
            if(n<=0)
                break;
            else{
                Pattern2 p2 = new Pattern2();
                p2.printPattern2(n);
            }
        }
    }
}